src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] map;
+    private int currentCapacity;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
-
+        updateThreshold();
+        if (size >= threshold) {
+            resize();
+        }
+        int index = generateIndex(key);
+        if (map[index] == null) {
+            map[index] = new Node<>(key, value);
+            size++;
```

Comment:
think what needs to be done to write the 'size++' once at the end of the method

Reply:
First if creates node in index, which was null. All following if's rewriting or adding node to list. I guess it's better to split it in two if statements and make 'size++' twice